code/modules/power/apc.dm

```diff
@@ -757,38 +757,47 @@
 	. = ..()
 	if(.)
 		return
-	var/mob/living/carbon/human/H = user
-	var/datum/species/ethereal/eth_species = H.dna?.species
-	if(istype(H) && istype(eth_species))
-		if(H.a_intent == INTENT_HARM)
-			if(eth_species.ethereal_charge >= ETHEREAL_CHARGE_FULL - 19)
-				to_chat(H, "<span class='warning'>Your charge is full!</span>")
+
+	if(isethereal(user))
+		var/mob/living/carbon/human/H = user
+		var/datum/species/ethereal/E = H.dna.species
+		if((H.a_intent == INTENT_HARM) && (E.drain_time < world.time))
+			if(cell.charge <= (cell.maxcharge / 2)) // if charge is under 50% you shouldn't drain it
+				to_chat(H, "<span class='warning'>The APC doesn't have much power, you probably shouldn't drain any.</span>")
 				return
-			if(cell.charge < 100)
-				to_chat(H, "<span class='warning'>The cell has no charge!</span>")
+			var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)

```

Comment:
If you can do organ manipulation on ethereal, this could runtime if their stomach is replaced for some ungodly reason